src/main/scala/treadle/TreadleTester.scala

```diff
@@ -74,8 +74,23 @@ class TreadleTester(annotationSeq: AnnotationSeq) {
 
   val startTime: Long = System.nanoTime()
 
+  private def findTopLevelClocks() = {
```

Comment:
I guess this was the simplest place to implement this feature, but it does lead to multiple slightly different mechanisms of dealing with clocks. As an alternative, you could add `FindClocks` pass to the `TreadlePhase` which will automatically generate the appropriate clock annotations. That way everyone uses clock annotations instead of having this alternative way inside the tester.
Not a blocker for me, but maybe something to consider.

Reply:
Leaving as is for now.